iceoryx_posh/source/popo/ports/client_port_roudi.cpp

```diff
@@ -38,23 +40,144 @@ ClientPortRouDi::MemberType_t* ClientPortRouDi::getMembers() noexcept
     return reinterpret_cast<MemberType_t*>(BasePort::getMembers());
 }
 
+QueueFullPolicy2 ClientPortRouDi::getResponseQueueFullPolicy() const noexcept
+{
+    return static_cast<QueueFullPolicy2>(getMembers()->m_chunkReceiverData.m_queueFullPolicy);
+}
+
 cxx::optional<capro::CaproMessage> ClientPortRouDi::tryGetCaProMessage() noexcept
```

Comment:
This is done in a similar fashion to the `PublisherPortRouDi` and `SubscriberPortRoudi` but I don't quite like it. The transitions of the state machine is done in two ways which makes reasoning about a correct implementation harder. I would suggest to rename this to `handlePendingStateTransitionsAndGetCaProResponse`. The logic of this method would be split into a `tryGetCaProMessageForStateTransition` method and the actual state transition moved into `dispatchCaProMessageAndGetPossibleResponse`. This method would then become just a thin wrapper around the other methods
```cpp
cxx::optional<capro::CaproMessage> handlePendingStateTransitionsAndGetCaProResponse() noexcept
{
    cxx::optional<capro::CaproMessage> response = nullopt;
    tryGetCaProMessageForStateTransition().and_then([&] (const auto& caProMessage) {
        response = dispatchCaProMessageAndGetPossibleResponse(caProMessage);
    });
    return response;
}
```